extension/test/data_test.ts

```diff
@@ -69,4 +69,209 @@ describe('data.ts', function () {
+        );
+
+        expect(singleKanjiDataEntryAsText).to.equal(
+          '<kanji-entry>\n<eigo-entry-1>; <eigo-entry-2>; <eigo-entry-3>\n<onkun-entry-1>、 <onkun-entry-2>、 <onkun-entry-3>\n名乗り\t<nanori-entry>\n部首名\t<bushumei-entry>\nHalpern\t<H-misc-entry>\nHeisig 5th Edition\t<L-misc-entry>\nHeisig 6th Edition\t<DN-misc-entry>\nHenshall\t<E-misc-entry>\nKanji Learners Dictionary\t<DK-misc-entry>\nKanji Learners Dictionary 2nd Edition\t<DL-misc-entry>\nNelson\t<N-misc-entry>\nNew Nelson\t<V-misc-entry>\nPinYin\t<Y-misc-entry>\nSkip Pattern\t<P-misc-entry>\nTuttle Kanji & Kana\t<IN-misc-entry>\nTuttle Kanji Dictionary\t<I-misc-entry>\nUnicode\t<U-misc-entry>\n'
+        );
+      });
+
+      it('even with large max, returns a single kanji data entry as text', function () {
+        const singleKanjiDataEntryAsText = rcxDict.makeText(
+          kanjiDictEntry,
+          /* max= */ 1000
+        );
+
+        expect(singleKanjiDataEntryAsText).to.equal(
+          '<kanji-entry>\n<eigo-entry-1>; <eigo-entry-2>; <eigo-entry-3>\n<onkun-entry-1>、 <onkun-entry-2>、 <onkun-entry-3>\n名乗り\t<nanori-entry>\n部首名\t<bushumei-entry>\nHalpern\t<H-misc-entry>\nHeisig 5th Edition\t<L-misc-entry>\nHeisig 6th Edition\t<DN-misc-entry>\nHenshall\t<E-misc-entry>\nKanji Learners Dictionary\t<DK-misc-entry>\nKanji Learners Dictionary 2nd Edition\t<DL-misc-entry>\nNelson\t<N-misc-entry>\nNew Nelson\t<V-misc-entry>\nPinYin\t<Y-misc-entry>\nSkip Pattern\t<P-misc-entry>\nTuttle Kanji & Kana\t<IN-misc-entry>\nTuttle Kanji Dictionary\t<I-misc-entry>\nUnicode\t<U-misc-entry>\n'
+        );
+      });
+
+      describe('empty fields', function () {
+        it('uses a hyphen after kanji field when eigo field is empty', function () {
+          const emptyEigoKanjiDictEntry = { ...kanjiDictEntry, eigo: '' };
+
+          const hyphenAsEigoEntryText = rcxDict.makeText(
+            emptyEigoKanjiDictEntry,
+            /* max= */ 1
+          );
+
+          expect(hyphenAsEigoEntryText).to.equal(
+            '<kanji-entry>\n-\n<onkun-entry-1>、 <onkun-entry-2>、 <onkun-entry-3>\n名乗り\t<nanori-entry>\n部首名\t<bushumei-entry>\nHalpern\t<H-misc-entry>\nHeisig 5th Edition\t<L-misc-entry>\nHeisig 6th Edition\t<DN-misc-entry>\nHenshall\t<E-misc-entry>\nKanji Learners Dictionary\t<DK-misc-entry>\nKanji Learners Dictionary 2nd Edition\t<DL-misc-entry>\nNelson\t<N-misc-entry>\nNew Nelson\t<V-misc-entry>\nPinYin\t<Y-misc-entry>\nSkip Pattern\t<P-misc-entry>\nTuttle Kanji & Kana\t<IN-misc-entry>\nTuttle Kanji Dictionary\t<I-misc-entry>\nUnicode\t<U-misc-entry>\n'
+          );
+        });
+
+        it('omits the nanori field after onkun field when nanori field is empty', function () {
+          const emptyNanoriKanjiDictEntry = { ...kanjiDictEntry, nanori: '' };
+
+          const noNanoriFieldKanjiEntryText = rcxDict.makeText(
+            emptyNanoriKanjiDictEntry,
+            /* max= */ 1
+          );
+
+          expect(noNanoriFieldKanjiEntryText).to.equal(
```

Comment:
For `omits` is some klind of `does.not.contain` better; though that wouldn't be as specific as including the relationship to onkun...